WombatProject/Controllers/BookshelvesController.cs

```diff
@@ -43,66 +40,28 @@ public async Task<ActionResult<Bookshelf>> GetBookshelf(int id)
             return bookshelf;
         }
 
-        // PUT: api/Bookshelves/5
-        // To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
-        [HttpPut("{id}")]
-        public async Task<IActionResult> PutBookshelf(int id, Bookshelf bookshelf)
-        {
-            if (id != bookshelf.Id)
-            {
-                return BadRequest();
-            }
-
-            _apiService.Context.Entry(bookshelf).State = EntityState.Modified;
-
-            try
-            {
-                await _apiService.Context.SaveChangesAsync();
-            }
-            catch (DbUpdateConcurrencyException)
-            {
-                if (!BookshelfExists(id))
-                {
-                    return NotFound();
-                }
-                else
-                {
-                    throw;
-                }
-            }
-
-            return NoContent();
-        }
-
         // POST: api/Bookshelves
-        // To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
         [HttpPost]
-        public async Task<ActionResult<Bookshelf>> PostBookshelf(Book book)
+        public async Task<ActionResult<Bookshelf>> AddItemToBookshelf(Book book)
         {
-            var bookshelf = await _apiService.AddBookToBookshelf(book);
+            var bookshelf = await _repository.AddBookToBookshelfAsync(book);
 
-            return CreatedAtAction("GetBookshelf", new { id = bookshelf.Id }, bookshelf);
+            return CreatedAtAction("GetBookshelfItemById", new { id = bookshelf.Id }, bookshelf);
         }
 
         // DELETE: api/Bookshelves/5
         [HttpDelete("{id}")]
-        public async Task<IActionResult> DeleteBookshelf(int id)
+        public async Task<IActionResult> RemoveBookFromBookshelf(int id)
```

Comment:
Method name can be: RemoveBookFromBookshelfById

Reply:
Yep, that's more consistent, it's done. :+1: